### PR TITLE
[SPARK-47498][TESTS][CORE] Refine some GPU fraction calculation tests

### PR DESCRIPTION
**Pull request by wbo4958**
### What changes were proposed in this pull request?

This PR refines some fractional GPU resource calculation tests.

### Why are the changes needed?

This PR adds more comments to the tests and refines some assertation blocks.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
The existing CI passes.

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by wbo4958** [comment]
Hi @Ngone51, Could you help review this PR, thx

File: core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala [hunk 1]
```diff
@@ -2654,18 +2659,21 @@ class TaskSchedulerImplSuite extends SparkFunSuite with LocalSparkContext
     taskScheduler.submitTasks(lowerTaskSet)
     taskScheduler.submitTasks(higherRpTaskSet)
 
-    // only offer the resources to the higher taskset
+    // Initially, offer the available resources to the higher priority task set, where each task
+    //      requires 0.7 GPU, so Spark can assign the GPU resources to a maximum of 4 tasks, leaving
+    //      0.3 GPU available for each remaining tasks.
+    // Secondly, try to offer the available resources to the lower priority task set, where each
+    //      task requires 0.4 GPU, while only 0.3 gpu of each address is available, so couldn't
+    //      offer any resources for the lower tasks.
```
Comment (Ngone51) [inline core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala:2667]: It looks `lowerTaskSet` is summitted before `higherRpTaskSet`? So it's actually requires 0.4 GPU per task first and then 0.7 GPU per task?

Comment (wbo4958) [reply core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala:2667]: LowerTaskSet is indeed submitted before higherRpTaskSet, but there's no resource to offer for lowerTaskSet at that time, lowerTaskSet will be put into the Pool (using FIFO algo by default). After both lowerTaskSet and higherRpTaskSet are in the Pool, the test case begins to offer resources and spark will choose the higher TaskSet (in this case, it is higherTaskSet) to offer. So it should behave like the comments.

Comment (wbo4958) [reply core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala:2667]: @Ngone51 Could you help review it again?

Comment (Ngone51) [reply core/src/test/scala/org/apache/spark/scheduler/TaskSchedulerImplSuite.scala:2667]: I see, thanks for the explanation.

**Review by Ngone51 — APPROVED** [review]

**Comment by Ngone51** [comment]
Thanks, merged to master!

**Pull request closed without merging**